src/cunumeric/index/repeat.cc

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/repeat.h"
+#include "cunumeric/index/repeat_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <LegateTypeCode CODE, int DIM>
+struct RepeatImplBody<VariantKind::CPU, CODE, DIM> {
+  using VAL = legate_type_of<CODE>;
+
+  size_t operator()(Buffer<VAL>& out,
+                    const AccessorRO<VAL, DIM>& in,
+                    const int64_t repeats,
+                    const int32_t axis,
+                    const Pitches<DIM - 1>& pitches,
+                    const Rect<DIM>& rect) const
+  {
```

Comment:
Just a note, the scalar case seems to also assume 1d arrays, so remember to update it in the future.